sm64-tas-scripting/GetMinimumDownhillWalkingAngle.cpp

```diff
@@ -44,9 +44,7 @@ bool GetMinimumDownhillWalkingAngle::execution()
 	*/
 
 	Object* marioObj = marioState->marioObj;
-
-	int pyramidID = 84;
-    Object* pyramidPlatform = (Object*)(game->addr("gObjectPool") + 1392*pyramidID);
+	Object* pyramidPlatform = marioState->floor->object;
```

Comment:
No issues here, but ideally we want to check that Mario's floor object is a pyramid in the verification() method, to guarantee that the script only executes in an appropriate situation. I do this in several other scripts, but I originally didn't for this one since the original implementation wasn't specific to the pyramid platform.

Reply:
That's a fair point. Though, my understanding is that verification() checks post-conditions for the script. But this is more of a pre-condition, rather than a post-condition, of the script. So maybe that needs to be handled differently. I guess you could add a separate method to the Script template that checks that all necessary pre-conditions for the script are met (and prevents execution if they're not).

Or, perhaps more simply, the functions that call GetMinimumDownhillWalkingAngle should check that before calling it, which I think is already the case. BitFsPyramidOscillation_RunDownhill does so implicitly (if Mario ever leaves the platform, the loop ends ). BitFsPyramidOscillation and BitFsPyramidOscillation_TurnThenRunDownhill don't check this, however the current versions of those scripts assume that Mario's current platform (at the frame the script is executed) is the pyramid, so if it's wrong in GetMinimumDownhillWalkingAngle then it'd be wrong in the parent script too.

Reply:
Currently verification() checks preconditions and validation() checks postconditions, but these names mean basically the same thing so it's kind of confusing. I think we should have validation() check preconditions and rename the current validation() method to assertion(). I can do that in a separate PR.

My intention with the script framework is that the scripts can be run in any situation, and so don't need to rely on the caller method using them responsibly. The precondition check is meant to ensure that even if the caller calls the script in an inappropriate situation, a meaningful result is achieved (i.e. game was not in a state adequate for the script and so no execution was performed).